src/util/json_expr.cpp

```diff
@@ -260,18 +274,33 @@ json_objectt json(
 
       bool is_signed=underlying_type.id()==ID_signedbv;
 
-      std::string sig=is_signed?"":"unsigned ";
-
-      if(width==config.ansi_c.char_width)
-        result["c_type"]=json_stringt(sig+"char");
-      else if(width==config.ansi_c.int_width)
-        result["c_type"]=json_stringt(sig+"int");
-      else if(width==config.ansi_c.short_int_width)
-        result["c_type"]=json_stringt(sig+"short int");
-      else if(width==config.ansi_c.long_int_width)
-        result["c_type"]=json_stringt(sig+"long int");
-      else if(width==config.ansi_c.long_long_int_width)
-        result["c_type"]=json_stringt(sig+"long long int");
+      if(mode==ID_C)
```

Comment:
This would introduce a dependency of `util` on `java_bytecode` which isn't good I think. But otherwise yes that would be better.